addons/account/models/account_journal.py

```diff
@@ -807,5 +807,6 @@ def _get_last_bank_statement(self, domain=None):
         '''
         self.ensure_one()
         last_statement_domain = (domain or []) + [('journal_id', '=', self.id)]
-        last_st_line = self.env['account.bank.statement.line'].search(last_statement_domain, order='date desc, id desc', limit=1)
-        return last_st_line.statement_id
+        last_st_lines = self.env['account.bank.statement.line'].search(last_statement_domain)
+        last_statement = self.env['account.bank.statement'].search([('id', 'in', last_st_lines.mapped('statement_id.id'))], order='date desc, id desc', limit=1)
+        return last_statement
```

Comment:
I'm not a huge fan of doing two searches here.
What about adapting the method to instead take a domain valid for the bank statement?

So instead of giving the method a domain `"move_id.state', '=', 'posted"`, it would be `"state', '=', 'posted"` or `"state', '=', 'confirmed"` I'm not sure which one is expected. (I guess posted)

And then you could simply do

```py
        self.ensure_one()
        last_statement_domain = (domain or []) + [('journal_id', '=', self.id)]
        last_statement = self.env['account.bank.statement'].search(last_statement_domain, limit=1)
        return last_statement
```

Reply:
Is it not changing the current behaviour ? As currently we search first for the statement lines which have move_id.state == 'posted' (not statement_id.state == 'posted')

Without changing the domain, I could do ```py
        self.ensure_one()
        last_statement_line_domain = (domain or []) + [('journal_id', '=', self.id)]
        last_st_lines = self.env['account.bank.statement.line'].search(last_statement_line_domain)
        return last_st_lines.statement_id[:1]```

Reply:
> Is it not changing the current behaviour ? As currently we search first for the statement lines which have move_id.state == 'posted' (not statement_id.state == 'posted')

I'm quoting LAS, about bank statements.

> if New, all lines must be in draft, if processing/confirmed, all lines must be posted

If we consider that, and use the domain [('state', 'in', ['posted', 'confirm'])] directly on the bank statement wouldn't we have the same behavior then? It will take the latest bank statement with posted lines (which is what we expect).

Reply:
Ok, ti makes sens now. thanks.